### PR TITLE
Class-based views

### PR DESCRIPTION
**Pull request by ghost**
Refactors (most) views to use Django's class-based generic CRUD views.

### CODE REVIEW

File: decisiontree/templates/tree/answer.html [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
-{% block content %}
 {% load i18n %}
+{% load tree_tags %}
+
+{% block content %}
 <form action="" method="POST" class="pretty"> {% csrf_token %}
 <fieldset>
 <legend>
-{% if answerid %}
+{% if object %}
```
Comment (vkurup) [inline decisiontree/templates/tree/answer.html:10]: Not related to this PR, but some of these templates could benefit from some indentation.

File: decisiontree/views/base.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+from django.contrib import messages
+from django.contrib.auth.decorators import login_required
+from django.db import transaction
+from django.utils.decorators import method_decorator
+from django.views.generic import DeleteView, DetailView, ListView, UpdateView
+from django.views.generic.detail import SingleObjectTemplateResponseMixin
+from django.views.generic.edit import ModelFormMixin, ProcessFormView
+
+
+def cbv_decorator(function_decorator):
```
Comment (vkurup) [inline decisiontree/views/base.py:12]: cool!

File: decisiontree/views/views.py [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+        tree = self.get_object()
+        if tree.has_loops():
+            return redirect('list-surveys')
+        all_states = tree.get_all_states()
+        output = StringIO()
+        w = csv.writer(output)
+        headings = ["Person", "Date"]
+        headings.extend([state.question for state in all_states])
+        w.writerow(headings)
+        sessions = models.Session.objects.all().filter(tree=tree)
```
Comment (vkurup) [inline decisiontree/views/views.py:141]: the `.all()` could be removed here.

File: decisiontree/views/views.py [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+        for session in sessions:
+            values = [str(session.connection), session.start_date]
+            transitions = map((lambda x: x.transition), session.entries.all())
```
Comment (vkurup) [inline decisiontree/views/views.py:144]: Could this also be `transitions = session.entries.values_list('transition', flat=True)`?

**Comment by vkurup** [comment]
Most of my comments above are unrelated to the actual PR (so feel free to ignore, or postpone). One additional unrelated thing is that we should add `south_migrations` to the excluded list in `.coveragerc`.

The actual changes in this PR look great. :tophat:

File: decisiontree/templates/tree/answer.html (hunk 1 above)
Comment (ghost) [inline decisiontree/templates/tree/answer.html:10]: Definitely. It's coming in another pull request.

File: decisiontree/views/views.py (hunk 3 above)
Comment (ghost) [inline decisiontree/views/views.py:141]: Done.

File: decisiontree/views/views.py (hunk 4 above)
Comment (ghost) [inline decisiontree/views/views.py:144]: Hmm, I don't think that would work - looks like the code uses the transition object, but `.values_list()` would return the primary key. 

This view didn't fit the basic logic that all the other ones did, so I didn't refactor it. I hope to be able to do so as we continue to update the app.

**Pull request merged**